### ISSUE TITLE
cannot access pry-doc from all directories

### ISSUE THREAD

**Issue by kaylee42** [pry/pry-doc#30]
I'm new to programming (which is why I got pry-doc in the first place!) so I'm sorry if my explanation is poor. I've attached a screenshot in the hopes this will help clarify. Basically, the issue I have is that when I drop into pry from certain directories I am unable to access the pry-doc (even after reinstalling the gem), but if I change directories it works just fine. My apologies if this is just something I'm doing wrong!

<img width="858" alt="screen shot 2015-12-17 at 10 41 11 am" src="https://cloud.githubusercontent.com/assets/14985351/11873881/bf9ffa74-a4aa-11e5-9c52-e82634e5d8fd.png">

**Comment by dgutov** [comment]
Looks for the file named `.ruby-version` in any of those directories.

**Comment by vanburg** [comment]
I have the same problem. @dgutov, please can you be more specific about this issue. I've tried both options with and without ``.ruby-version`` (with pry reloading of course) and both failed. I believe the last option does not matter, though (there's ``RUBY_VERSION`` const for that, and not every project use ``.ruby-version``).

**Comment by dgutov** [comment]
If `.ruby-version` is present, it dictates which version of Ruby is used (if you're using rbenv/rvm/etc, of course). And that, in turn, affects which gems are considered installed. Try `gem install pry-doc` from that directory.

And if that still doesn't work, this bug report will need more information, such as full directory names.

**Comment by dgutov** [comment]
And `ruby --version` output.

**Comment by vanburg** [comment]
@dgutov thanks for you attention!
Here are the steps reproducing the problem:
```bash
vanburg@vb-mac ~/Sites $ pry -v
Pry version 0.10.3 on Ruby 2.3.0

vanburg@vb-mac ~/Sites $ gem list | grep pry-doc
pry-doc (0.8.0)

# No .ruby-version here
vanburg@vb-mac ~/Sites $ ls | grep ruby-v

vanburg@vb-mac ~/Sites $ pry
[1] pry(main)> ? loop

From: vm_eval.c (C Method):
Owner: Kernel
Visibility: private
Signature: loop()
Number of lines: 12

Repeatedly executes the block.

If no block is given, an enumerator is returned instead.

   loop do
     print "Input: "
     line = gets
     break if !line or line =~ /^qQ/
     # ...
   end

StopIteration raised in the block breaks the loop.

# -------------------

vanburg@vb-mac ~/Sites $ cd prox_pass

vanburg@vb-mac ~/Sites/prox_pass$ cat .ruby-version
2.2.1

vanburg@vb-mac ~/Sites/prox_pass$ pry -v
Pry version 0.10.1 on Ruby 2.2.1

vanburg@vb-mac ~/Sites/prox_pass$ gem list | grep pry-doc
pry-doc (0.8.0)

vanburg@vb-mac ~/Sites/prox_pass$ pry

[1] pry(main)> ? loop
Error: Cannot locate this method: loop. Try `gem-install pry-doc` to get access to Ruby Core documentation.
```
Removing ``.ruby-version`` doesn't change this strange behaviour...

**Comment by dgutov** [comment]
In the latter case you have Pry 0.10.1. What if you upgrade to 0.10.3?

I've tried Pry 0.10.1 with PryDoc 0.6.0 and Pry 0.10.3 with PryDoc 0.6.0 and 0.8.0 (only because I upgraded Pry first), all using Ruby 2.2.2. All cases worked fine.

So before investigating deeper, I'd suggest you update to Pry 0.10.3 and/or Ruby 2.2.2, and try again.

**Comment by vanburg** [comment]
Nope, this doesn't help either:
```sh
pry -v
Pry version 0.10.3 on Ruby 2.2.1

[1] pry(main)> ? loop
Error: Cannot locate this method: loop. Try `gem-install pry-doc` to get access to Ruby Core documentation.
```

**Comment by dgutov** [comment]
Sigh. You're using RVM, I use rbenv. Care to try the latter?

**Comment by dgutov** [comment]
Oh, and try installing Ruby 2.2.2 first, of course.

**Comment by alankis** [comment]
Hi folks, is this issue actually closed or not? I have similar issue with rbenv and pry-doc. #33

**Comment by dgutov** [comment]
@alankis Things are just as they seem.

**Comment by agrberg** [comment]
I was also having this issue. I could access docs when outside of my current project's folder but not within it. My current project is a Rails so I tried adding it to the Gemfile and it now works.

For reference here is my current setup:
package | version
-------- | -------
rvm | 1.27.0
ruby | 2.3.0p0
Rails | 4.2.6
Pry | 0.10.3
pry-doc | 0.9.0

**tl;dr** if you are on Rails, see if adding `gem 'pry-doc'` to your `Gemfile` fixes the issue.

**Comment by dgutov** [comment]
> if you are on Rails, see if adding gem 'pry-doc' to your Gemfile fixes the issue

Or even if you're just using Bundler. If you're aware how it works (or why), that should be fairly obvious.

But I'm not sure it's the problem the OP was having.

**Comment by nafaabout** [comment]
I got this issue with a Rails app that's running on ruby 2.5.3. Adding pry-doc to Gemfile fixed the issue.

**Comment by andrehjr** [comment]
If you're still hitting this issue, make sure you are using an updated version of pry-doc. And file a new issue if the issue persists.

It seems working just fine for me with or without Rails. So I'll just close this for now 🙌

**Comment by andrehjr** [comment]
btw, on a vanilla pry session, you do need to call `require 'pry-doc'` in order for it to work.

When adding it on the Gemfile it gets required via Bundler so it 'just works'.
